src/shogun/mathematics/linalg/LinalgNamespaceMatrix.cpp

```diff
@@ -0,0 +1,43 @@
+#include <shogun/lib/DataType.h>
+#include <shogun/mathematics/linalg/LinalgNamespaceMatrix.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+
+#define DEFINE_SWITCH(PTYPE, FUNC, ...) \
+switch (PTYPE) \
+{ \
+	case PT_FLOAT32: \
+		FUNC<float32_t>(__VA_ARGS__); \
```

Comment:
+1 for the lazy, though quite a bit of API work needed for that. Linalg currently cannot do that.
So whats the best way forward right now in here?